templates/install_puppet.bat.erb

```diff
@@ -1,22 +1,36 @@
+SET
+
 set AGENT_PID=%1
 set windowTitle=Puppet Agent Upgrade
 title %windowTitle%
 
 set pid=
 for /f "tokens=2" %%a in ('tasklist /v ^| findstr /c:"%windowTitle%"') do set pid=%%a
-set pid_path=%TEMP%\puppet_agent_upgrade.pid
+set pid_path=%~dp0puppet_agent_upgrade.pid
```

Comment:
Comment withdrawn.  I was mistakenly thinking that `~dp0` is the template directory.  Its the location of the `install_puppet.bat` created from the ERB template, which is known to be the temp directory.

👍 to the change.